Orange/widgets/visualize/owscatterplot.py

```diff
@@ -380,8 +380,14 @@ def update_graph(self, attributes=None, reset_view=True, **_):
         self.graph.zoomStack = []
         if attributes and len(attributes) == 2:
             self.attr_x, self.attr_y = attributes
+        X = BasicStats(self.data[:, self.attr_x], self.attr_x)
+        Y = BasicStats(self.data[:, self.attr_y], self.attr_y)
```

Comment:
I think [`set_data` is the callback](https://github.com/biolab/orange3/pull/1096/files#diff-0f1cb6d8751836f98c5293bec4004169R45) called when something new is sent on the data signal.